errors/errctx_test.go

```diff
@@ -0,0 +1,155 @@
+package errors
+
+import (
+	"context"
+	stdErrors "errors"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-utils/logger"
+)
+
+func TestErrCtx_GetDeepestContext(t *testing.T) {
+	t.Run("It should return a nil context if the error is not wrapped with ErrCtx", func(t *testing.T) {
+		// Given
+		err := stdErrors.New("main error")
+		err = errgo.Notef(err, "wrapping error in func2")
+
+		// When
+		deepestCtx := GetDeepestContext(err)
+
+		// Then
+		assert.Nil(t, deepestCtx)
+		assert.Contains(t, err.Error(), "wrapping error in func2")
+	})
+
+	t.Run("It should get the deepest context and contains fields from multiple wrapped error", func(t *testing.T) {
+		// Given
+		ctx, _ := logger.WithFieldToCtx(context.Background(), "field0", "value0")
```

Comment:
Could you not use the `log` libs, and use `context.Withvalue` ? from the `context` package? (it would prevent casting below)